Articles: Git.md

Rafase282 opened the pull request:
Closes #875

theflametrooper commented:
LGTM :+1:

theflametrooper commented:
Closes #8510

abhisekp commented on .github/CONTRIBUTING.md:
```diff
@@ -3,7 +3,7 @@
 - [How To Contribute To The Wiki:](https://github.com/FreeCodeCamp/FreeCodeCamp/wiki/Wiki-Contribute-Local-GUI) This is a guide that will show you how to have a local copy so you can make changes from your computer, it also works for contributing to the main repository.
 - [More on how to make a pull request from a fork:](https://github.com/FreeCodeCamp/FreeCodeCamp/wiki/Pull-Request-Contribute) This guide has additional information on Pull requests in general and works for contributing to the main repository.
-- [How to Squash Commits:](https://github.com/FreeCodeCamp/FreeCodeCamp/wiki/git-rebase#squashing-multiple-commits-into-one) We require only one commit per Pull Request unless otherwise specified by a moderator. Please go through this guide to squash multiple commits.
+- [How to Squash Commits:](https://github.com/FreeCodeCamp/FreeCodeCamp/wiki/Git-Squash) We require only one commit per Pull Request unless otherwise specified by a moderator. Please go through this guide to squash multiple commits.
```
```diff
- [How to Squash Commits:](https://github.com/FreeCodeCamp/FreeCodeCamp/wiki/Git-Squash)
+ [How to Squash Commits:](Git-Squash)
```

abhisekp replied:
```diff
- [How to Squash Commits:](https://github.com/FreeCodeCamp/FreeCodeCamp/wiki/Git-Squash)
+ [How to Squash Commits:](Git-Squash)
```

abhisekp commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -24,7 +24,7 @@ Ensure that you use markdown syntax for articles, we do not accept `HTML` or `CS
 **Commits and Squashing:**
 - Your Pull Request/Submission should contain one single commit.
-- For multiple commits follow the [guide to squash them](git-rebase#squashing-multiple-commits-into-one).
+- For multiple commits follow the [guide to squash them](Git-Squash).
```
```diff
- [guide to squash them](Git-Squash)
+ [guide to squash them](https://github.com/FreeCodeCamp/FreeCodeCamp/wiki/Git-Squash)
```

That relative URL won't work but this will `[guide to squash them](../Git-Squash)` but it might conflict with the camperbot regex. Please confirm @Rafase282

Rafase282 replied:
Why wouldn't the relative link work?

abhisekp replied:
@Rafase282 oops! I remember removing the comment right after making the comment. But seems like it didn't remove. Sorry!

The relative will work. :+1:

abhisekp commented on Git-Rebase.md:
```diff
@@ -1,4 +1,6 @@
-**Were you referred to this page to get your PR accepted? Scroll right [to the bottom](git-rebase#git-rebase-at-fcc) then!**
+# Git rebase
+
+**Were you referred to this page to get your PR accepted? Scroll right [to the bottom](Git-Rebase#Git-Rebase-at-fcc) then!**
```
```diff
- [to the bottom](Git-Rebase#Git-Rebase-at-fcc)
+ [to the bottom](#git-rebase-at-FCC)
```

abhisekp commented:
For links, I would prefer something like this :point_right: [**Practice Modern JavaScript README**](https://github.com/abhisekp/Practice-Modern-JavaScript#app-top) i.e. with *`<a>` (anchor)* tags. 

Also see [**Airbnb Guide**](https://github.com/airbnb/javascript)

abhisekp commented on Git-Rebase.md:
```diff
@@ -35,7 +37,7 @@ The repo will now look like this:
 
 
-## git-rebase at FCC
+## Git-Rebase at FCC
```
Something like this would make a perma link
```md
## <a id="fcc-git-rebase"></a>Git-Rebase at FCC
```
which will appear as...

## <a id="fcc-git-rebase"></a>Git-Rebase at FCC

abhisekp commented on Git-Shortcut.md:
```diff
@@ -1,10 +0,0 @@
-#Git shortcut
```
This article can be improved to show some common git shortcuts. Why delete it?

Rafase282 replied:
Because of Git-Aliases it was rename.

abhisekp replied:
Oh! LOL. ok. :+1:

abhisekp commented on Git-Squash.md:
````diff
@@ -0,0 +1,22 @@
+
+That will open up a text-editor with something similar to the following:
+
+```
````
\`\`\`text

Rafase282 replied:
Without looking at the markdown, what's the difference.

```text
git rebase -i HEAD~n
```

```
git rebase -i HEAD~n
```

abhisekp replied:
I didn't mean syntax highlight for that although those should also be properly formatted :grin: 
```text
    ```sh
    $ git rebase -i HEAD~n
    ```
```

I meant for these 
```
    ```text
    pick commit_1
    pick commit_2
    ```
```

Rafase282 replied:
It is still the same, the is not highlight and they don't help with SEO so I don't see the point.

 ```text
    pick commit_1
    pick commit_2
 ```

 ```
    pick commit_1
    pick commit_2
 ```

abhisekp commented on Git-Squash.md:
````diff
@@ -0,0 +1,22 @@
+pick commit_n
+# Bunch of comments
+```
+
````
Also add a line here suggesting to read the instructions given in the text editor at the bottom.

abhisekp commented on Git-Squash.md:
```diff
@@ -0,0 +1,22 @@
+
+Leave the first commit alone, and change the rest of the `pick`s to `squash`. Save and exit the editor.
+
+If you've already pushed to a remote before squashing your commits, you'll have to push to the remote again, with the `-f` flag, otherwise git will throw an error at you. Happy coding!
```
"throwing error and Happy Coding" doesn't flow in a single line :wink: 

*Happy Gitting!* :smile:

abhisekp commented on Git.md:
````diff
@@ -1,36 +1,47 @@
+
+```
+$ git --distributed-is-the-new-centralized
+```
````
Please remove this completely.
```
    ```
    $ git --distributed-is-the-new-centralized
    ```
```

abhisekp commented on Git.md:
````diff
@@ -1,36 +1,47 @@
+$ git --distributed-is-the-new-centralized
+```
+
+_Git is a free and open source distributed version control system designed to handle everything from small to very large projects with speed and efficiency._
````
Un-emphasize this

abhisekp commented on Git.md:
```diff
@@ -1,36 +1,47 @@
+
+## Resources
+
+- [Ammend Last](Git-Amend-Last)
```
~~Ammend~~ -> Amend

abhisekp commented on Git.md:
```diff
@@ -1,36 +1,47 @@
+- [Clone a specific branch](Git-Clone-Specific-Branch)
+- [Delete a branch both locally and remotely](Git-Delete-a-Branch-both-locally-and-remotely)
+- [Discard unstaged changes](Git-Discard-Unstaged-Changes)
+- [Force git to overwrite local files on pull](Git-Force-Git-to-overwrite-local-files-on-pull) 
```
The file **`Git-Force-Git-to-overwrite-local-files-on-pull.md`** needs to be renamed to **`Git-Pull-Overwrite-Local-Files.md`**

and thus

```diff
- [Force git to overwrite local files on pull](Git-Force-Git-to-overwrite-local-files-on-pull) 
+ [Force Git to Overwrite Local Files on pull](Git-Pull-Overwrite-Local-Files) 
```

abhisekp commented on Git.md:
```diff
@@ -1,36 +1,47 @@
+- [Undo Git Add](Git-Undo-Git-Add)
+- [Undo and Redo A Commit](Git-Undo-Redo-A-Commit)
+- [Untrack Files](Git-Untrack-Files)
+
```
> **NOTE: ALL THE ABOVE LINKS MUST BE CHECKED AFTER THIS PR IS MERGED**

abhisekp commented on Git.md:
```diff
@@ -1,36 +1,47 @@
+
+_Git is a free and open source distributed version control system designed to handle everything from small to very large projects with speed and efficiency._
+
+## Resources
```
I think, this can be written as

## Articles

instead of

## Resources

```diff
- ## Resources
+ ## Articles
```

abhisekp commented:
@Rafase282 **File Rename**
```diff
- How-To-Create-A-Pull-Request-for-Free-Code-Camp.md
+ FCC-Pull-Request.md
```

abhisekp commented on Wiki-Contribute-Online.md:
```diff
@@ -38,7 +38,7 @@ Next is to click on `Create Pull Request`. it is the green button and it will ta
 The first thing you will notice is the title with the pull request number, in the example it is `#393`. Next, is the status of the pull request, it will either be `Open`, `Closed` or `Merged`.
 
-You can click on the tabs to view the commits. Always try to have everything in one commit, or as few as possible if you have a good reason to have more than one commit. If you have multiple commits and need to **squash** them then follow these [instructions](https://github.com/freecodecamp/freecodecamp/wiki/git-rebase#squashing-multiple-commits-into-one).
+You can click on the tabs to view the commits. Always try to have everything in one commit, or as few as possible if you have a good reason to have more than one commit. If you have multiple commits and need to **squash** them then follow these [instructions](https://github.com/freecodecamp/freecodecamp/wiki/Git-Squash).
```
```diff
- [instructions](https://github.com/freecodecamp/freecodecamp/wiki/Git-Squash)
+ [instructions](Git-Squash)
```

abhisekp commented on kr.lang/KR-Home.md:
```diff
@@ -27,7 +27,7 @@ JavaScript와 연관되거나 특정 커리큘럼이 아니더라도, **기술 
 - [Wiki에 기여하는 방법](https://github.com/FreeCodeCamp/wiki/blob/master/Wiki-Contribute-Local-GUI): 당신의 컴퓨터로부터 변경할 수 있도록 로컬 복사본을를 가지고 오는 방법을 보여주는 가이드이며, 메인 저장소에 기여하기 위해 작동합니다.  
 - [Fork로부터 pull request를 만드는 방법](https://github.com/FreeCodeCamp/wiki/blob/master/Pull-Request-Contribute) 
-- [Squash Commit하는 방법](https://github.com/FreeCodeCamp/wiki/blob/master/git-rebase#squashing-multiple-commits-into-one): 관리자에 의해 별도로 지정되지 않으면 Pull Request 당 오직 하나의 commit이 필요합니다. 
+- [Squash Commit하는 방법](https://github.com/FreeCodeCamp/wiki/blob/master/Git-Squash): 관리자에 의해 별도로 지정되지 않으면 Pull Request 당 오직 하나의 commit이 필요합니다. 
```
```diff
- [Squash Commit하는 방법](https://github.com/FreeCodeCamp/wiki/blob/master/Git-Squash)
+ [Squash Commit하는 방법](Git-Squash)
```

abhisekp commented:
@Rafase282 My Review is Complete.
Please implement the suggestions and add commits.

> **PLEASE DON'T FORCE PUSH. 
ADD MORE COMMITS ON TOP**

Rafase282 commented:
Suggestions implemented.

abhisekp commented:
@Rafase282 some of the suggestions are not yet complete although completely unnecessary for how it renders but just for the sake of consistency. 

- https://github.com/FreeCodeCamp/wiki/pull/879#discussion_r62457508
- https://github.com/FreeCodeCamp/wiki/pull/879#discussion_r62537756

and what is your opinion about this :point_right: https://github.com/FreeCodeCamp/wiki/pull/879#issuecomment-217791207

Rafase282 commented:
I already added the syntax highlight and I did add the line at the bottom but also added the sample squashing message so they see all the options that comes with it.

With regard tot he links I didnt see somethign clear and i was short on time. Also we should stick to something because whatever we use, we will need to implement that on the generator and currently I think relative links is easier since they work ont eh regular wiki and are easier to handle as the file is the same and just the root url needs to be changed/added and that is something that needs to be fixed on the generator.

abhisekp commented on Git-Rebase.md:
````diff
@@ -1,71 +1,54 @@
+```
+git checkout auth
+git rebase dev
+```
````
> **BONUS** :100: : *Not related to this PR.*
>  
  ```
      ```sh
      $ git checkout auth
      $ git rebase dev
      ```
  ```

Rafase282 replied:
Done.

abhisekp commented on Git-Rebase.md:
```diff
@@ -1,71 +1,54 @@
+
+If you contribute to the FCC codebase, or are planning to do it, always run this command before you make any changes in your local files and push them:
+
+`git pull --rebase upstream staging` (thanks, @SaintPeter)
```
> **BONUS :** :100: _not related to this PR_
```diff
- git pull --rebase upstream staging` (thanks, @SaintPeter)
+ git pull --rebase=preserve upstream staging`
```

Rafase282 replied:
It says to run that before making any changes, so assuming that preserve is to keep your changes, that would not be needed I think.

abhisekp commented on Git-Squash.md:
````diff
@@ -0,0 +1,42 @@
+# However, if you remove everything, the rebase will be aborted.
+#
+# Note that empty commits are commented out
+```
````
@Rafase282 I think adding instructions of the editor is too much.

Lets just say. "Also, you may read the instructions in the text editor"

abhisekp commented:
# :+1: LGTM

Thanks for this one. :tada: :smiley: 

----
### Merging in 5 minutes

If anyone wants to STOP me, please comment.
> # STOP Merging :hand: 
```md
# STOP Merging :hand: 
``

abhisekp commented:
# PR Merged :+1:
